routes/project/location/private_subnet.rb

```diff
@@ -35,10 +40,15 @@ class Clover
         r.post "disconnect", String do |disconnecting_ps_ubid|
           authorize("PrivateSubnet:disconnect", ps.id)
           subnet = PrivateSubnet.from_ubid(disconnecting_ps_ubid)
+          unless subnet
+            response.status = 400
+            next {error: {message: "Subnet to be disconnected not found"}}
+          end
+
           authorize("PrivateSubnet:disconnect", subnet.id)
           ps.disconnect_subnet(subnet)
           flash["notice"] = "#{subnet.name} will be disconnected in a few seconds"
-          ""
+          r.redirect "#{@project.path}#{ps.path}"
```

Comment:
Since we're sending an AJAX request, the returned value isn't necessary. Therefore, we can return `204`.

Reply:
AJAX request just refreshes the page

Reply:
Are the last changes OK?